learn/getting-started/building-a-data-service/building-a-data-service.md

```diff
@@ -0,0 +1,323 @@
+In addition, this tutorial will also elaborate on how you can create an HTTP RESTful API using Ballerina that can be used
+by end-users to perform basic CRUD operations on the database.
+
+![Data Service Architecture](/learn/images/data-service-architecture.png "Data Service Architecture")
+
+The methodology outlined in this tutorial can be used to work with PostgreSQL, SQL Server, OracleDB or any other 
+relational database as well using the [`PostgreSQL`](https://central.ballerina.io/ballerinax/postgresql),
+[`MSSQL`](https://central.ballerina.io/ballerinax/mssql), [`OracleDB`](https://central.ballerina.io/ballerinax/oraceldb)
+and [`JDBC`](https://central.ballerina.io/ballerinax/java.jdbc) connectors for Ballerina respectively.
+
+## Prerequisites
+
+### Setting up a MySQL Server Instance
+
+There are several methods you can use to set up a MySQL server.
+1. Install a MySQL server on your machine locally by downloading and installing MySQL as detailed on the [MySQL website](https://dev.mysql.com/doc/mysql-getting-started/en/#mysql-getting-started-installing) for different platforms.
+2. Using a cross-platform web-server solution such as [XAMPP](https://www.apachefriends.org/index.html) or [WampServer](https://www.wampserver.com/en/).
+3. Using [Docker](https://dev.mysql.com/doc/mysql-installation-excerpt/8.0/en/docker-mysql-getting-started.html) to create a MySQL server deployment.
+4. Using a cloud-based MySQL solution such as Google’s [CloudSQL](https://cloud.google.com/sql), Amazon’s [RDS for MySQL](https://aws.amazon.com/rds/sqlserver/), or Microsoft’s [Azure Database for MySQL](https://azure.microsoft.com/en-us/services/mysql/).
+
+>Be sure to note down the connection and authentication details for connecting to the MySQL server including the hostname, port, username and password.
+
+### Creating a Database and Table
+
+Connect to the MySQL server using the terminal (or any other preferred method) and execute the following commands to 
+create a database and table. This tutorial will demonstrate the basic use-case of creating, maintaining and
+interacting with a database of employees in an organization.
+
+```roomsql
+CREATE DATABASE IF NOT EXISTS Company;
+
+CREATE TABLE Company.Employees (
+	employee_id INTEGER AUTO_INCREMENT PRIMARY KEY,
+	first_name  VARCHAR(255) NOT NULL,
+	last_name   VARCHAR(255) NOT NULL,
+	email       VARCHAR(255) NOT NULL,
+	phone       VARCHAR(50) NOT NULL ,
+	hire_date   DATE NOT NULL,
+	manager_id  INTEGER REFERENCES Employees(employee_id),
+	job_title   VARCHAR(255) NOT NULL
+)
+```
+
+### Creating a Ballerina Project
+
+A new Ballerina project can be created by executing the following command in the directory in which you want to create the project.
+
+```shell
+bal new data_service
+``` 
+
+>For more information on creating Ballerina packages, see [Creating your First Ballerina Package](/learn/hello-world/creating-your-first-ballerina-package/#creating-the-package/).
```

Comment:
```suggestion
>For more information on creating Ballerina packages, see [Creating your First Ballerina Package](/learn/hello-world/creating-your-first-ballerina-package/#creating-the-package/).
```
Please correct the format of the info, tips, and note sections in all occurrences.